pytensor/tensor/slinalg.py

```diff
@@ -64,21 +65,52 @@ def make_node(self, x):
         dtype = scipy_linalg.cholesky(np.eye(1, dtype=x.type.dtype)).dtype
         return Apply(self, [x], [tensor(shape=x.type.shape, dtype=dtype)])
 
+    def _cholesky(
+        self, a, lower=False, overwrite_a=False, clean=True, check_finite=False
+    ):
+        a1 = np.asarray_chkfinite(a) if check_finite else np.asarray(a)
```

Comment:
We can trust the input to always be numpy arrays in our case, which gets rid of `np.asarray` and the `_datacopied` below